src/main/java/org/openstreetmap/atlas/checks/validation/intersections/IntersectingBuildingsCheck.java

```diff
@@ -159,8 +152,28 @@ public boolean validCheckForObject(final AtlasObject object)
             // Flag based on intersection type
             if (resultType == IntersectionType.OVERLAP)
             {
-                flag.addObject(otherBuilding, this.getLocalizedInstruction(0,
-                        object.getOsmIdentifier(), otherBuilding.getOsmIdentifier()));
+                // Get object as a Surface
+                final Surface objectAsSurface = ((Area) object).asPolygon().surface();
+                // If object is larger than otherBuilding, the instruction states object contains
+                // otherBuilding
+                if (objectAsSurface.isLargerThan(otherBuilding.asPolygon().surface()))
```

Comment:
Every call to `otherBuilding.asPolygon().surface()` method will create a polygon and calculate surface area again. Please cache the value to avoid redundant calculations.